safe_eth/eth/account_abstraction/bundler_client.py

```diff
@@ -172,14 +178,21 @@ def get_user_operation_and_receipt(
             self._get_user_operation_by_hash_payload(user_operation_hash, request_id=1),
             self._get_user_operation_receipt_payload(user_operation_hash, request_id=2),
         ]
-        result = self._do_request(payload)
-        if not (result and result[0]):
+        results = self._do_request(payload)
+        if (
+            results
+            and isinstance(results, list)
+            and results[0] is not None
+            and isinstance(results[0], dict)
```

Comment:
```suggestion
            and isinstance(results[0], dict)
```

Shouldn't this be enough for `mypy`?

Reply:
It's enough, but you need to check if results have at least two elements to avoid `list index out of range`. Updated [here](https://github.com/safe-global/safe-eth-py/pull/1308/commits/1d71d495526362616735c2fdb81c507054333525).

Reply:
Nice